src/ralph/assets/subscribers.py

```diff
@@ -0,0 +1,128 @@
+    """
+    business_owners = [
+        get_user_model().objects.get_or_create(username=i['username'])[0]
+        for i in business_owners
+    ]
+    technical_owners = [
+        get_user_model().objects.get_or_create(username=i['username'])[0]
+        for i in technical_owners
+    ]
+
+    service.business_owners = business_owners
+    service.technical_owners = technical_owners
+
+
+def _update_service_environments(service, environments):
+    """
+    Update service environments.
+
+    Args:
+        service: Service
+        environments: list of environment names
+    """
+    new_envs = []
+    for env_name in environments:
+        new_envs.append(Environment.objects.get_or_create(name=env_name)[0])
+
+    current = set([u.id for u in service.environments.all()])
+    new = set([u.id for u in new_envs])
+    to_delete = current - new
+    to_add = new - current
+    for env_id in to_add:
+        ServiceEnvironment.objects.create(
+            service=service,
+            environment_id=env_id
+        )
+    for env_id in to_delete:
+        ServiceEnvironment.objects.get(
+            service=service, environment_id=env_id
+        ).delete()
+
+
+@pyhermes.subscriber(topic=settings.HERMES_SERVICE_TOPICS['CREATE'])
+@pyhermes.subscriber(topic=settings.HERMES_SERVICE_TOPICS['UPDATE'])
+@pyhermes.subscriber(topic=settings.HERMES_SERVICE_TOPICS['REFRESH'])
+def update_service_handler(data):
+    """
+    Update Service data from hermes.
+
+    Example data structures:
+        {
+            'uid': 'service uid',
+            'name': 'service name',
+            'status': 'service status',
+            'isActive': 'boolean if service is active',
+            'environments': 'service environments',
+            'businessOwners': [
+                {'username': 'username'}
+            ],
+            'technicalOwners': [
+                {'username': 'username'}
+            ]
+        }
+    """
+    try:
+        service, _ = Service.objects.update_or_create(
+            uid=data['uid'],
+            defaults={
+                'name': data['name'],
+                'active': data['isActive'],
+            }
+        )
+    except Exception as e:
+        logger.exception(e)
+    else:
+        _update_service_owners(
+            service=service,
+            business_owners=data['businessOwners'],
+            technical_owners=data['technicalOwners']
+        )
+        _update_service_environments(
+            service=service,
+            environments=data['environments']
+        )
+
+        logger.info(
+            'Synced service `{}` with UID `{}`.'.format(
+                data['name'], data['uid']
+            ),
+            extra=data
+        )
+
+
+@pyhermes.subscriber(topic=settings.HERMES_SERVICE_TOPICS['DELETE'])
+def delete_service_handler(data):
+    """
+    Set service active to False if service deleted.
+    """
+    try:
+        Service.objects.filter(uid=data['uid']).update(active=False)
```

Comment:
check first if there is anything related to this service (for example any BaseObject) - if yes, then report an error and don't deactivate it